### PR TITLE
enable proxy for Stripe HTTP connection

### PR DESCRIPTION
**Pull request by xbwu**
follow steps to tunnel Stripe connection through a proxy
```
Stripe.setConnectionProxy(<java.net.Proxy>)

// if proxy server requires authorization
Stripe.setProxyCredential(<java.net.PasswordAuthentication>)
```

### CODE REVIEW

File: src/main/java/com/stripe/net/LiveStripeResponseGetter.java [hunk 1]
```diff
@@ -118,7 +119,18 @@ private static String urlEncodePair(String k, String v)
...
 			stripeURL = new URL(url);
 		}
-		java.net.HttpURLConnection conn = (java.net.HttpURLConnection) stripeURL.openConnection();
+		HttpURLConnection conn;
+		if (Stripe.getConnectionProxy() != null) {
+			conn = (HttpURLConnection) stripeURL.openConnection(Stripe.getConnectionProxy());
+			Authenticator.setDefault(new Authenticator() {
+				@Override
+				protected PasswordAuthentication getPasswordAuthentication() {
+					return Stripe.getProxyCredential();
```
Comment (kyleconroy) [inline src/main/java/com/stripe/net/LiveStripeResponseGetter.java]: Is it okay for `PasswordAuthentication` to return nil?

Comment (xbwu) [inline src/main/java/com/stripe/net/LiveStripeResponseGetter.java]: yes, the default impl is returning a null PasswordAuthentication. and i've verified.
```
scala> import java.net._
import java.net._
scala> import com.stripe._
import com.stripe._
scala> import com.stripe.model._
import com.stripe.model._
scala> val m = new java.util.HashMap[String, Object]
m: java.util.HashMap[String,Object] = {}
scala> m.put("description", "xbwu.enable.proxy.test")
res0: Object = null
scala> Stripe.apiKey="sk_test_..."
Stripe.apiKey: String = sk_test_...
scala> Stripe.setConnectionProxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("localhost", 3128)))
scala> Customer.create(m)
com.stripe.exception.APIConnectionException: IOException during API request to Stripe (https://api.stripe.com): Unable to tunnel through proxy. Proxy returns "HTTP/1.1 407 Proxy Authentication Required" Please check your internet connection and try again. If this problem persists,you should check Stripe's service status at https://twitter.com/stripestatus, or let us know at support@stripe.com.
scala> Stripe.setProxyCredential(new PasswordAuthentication("payment", "......".toCharArray))
scala> Customer.create(m)
res12: com.stripe.model.Customer =
<com.stripe.model.Customer@1074821682 id=cus_6pcgy8R2GZ0D7w> JSON: {......}
scala> Stripe.setProxyCredential(null)
scala> Stripe.setConnectionProxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("192.168.199.199", 3128)))
scala> Customer.create(m)
res29: com.stripe.model.Customer =
<com.stripe.model.Customer@589773472 id=cus_6pclJkgE5yM3N6> JSON: {......}
```
in my environment, localhost:3128 is tunneling to a proxy server which requires authorization, while 192.168.199.199:3128 is a proxy server doesn't need auth

**Comment by kyleconroy** [comment]
@xbwu Fantastic.

**Comment by xbwu** [comment]
thanks @kyleconroy, when should i expect new version being published to maven

**Comment by kyleconroy** [comment]
@xbwu today or tomorrow. I'll ping here once it's released.

**Comment by xbwu** [comment]
awesome, thanks @kyleconroy

**Pull request merged**